core/src/cluster_info.rs

```diff
@@ -471,8 +483,12 @@ impl ClusterInfo {
         *self.entrypoint.write().unwrap() = Some(entrypoint)
     }
 
-    pub fn id(&self) -> Pubkey {
-        self.id
+    pub fn identity(&self) -> Pubkey {
+        self.id_keypair.pubkey()
+    }
+
+    pub fn node_id(&self) -> Pubkey {
+        self.identity
```

Comment:
I think it's a bit confusing that `identity()` doesn't return `self.identity`, while `node_id` returns `self.identity`.

How about 

`identity -> gossip_identity`,
`node_id() -> gossip_identity()`
`identity() -> validator_id()`
`id_keypair -> validator_keypair`

Reply:
Totally agree, actually the only reason It's like this is I renamed the original `id()` method to `identity()` and wrapped this one with `node_id` specifically so I could make sure while I was going through errors in the source I would be able to identify what is using which key. I used it as a trick so I wouldn't have to keep a mental note, and haven't undone it while I play with the changes locally.

I'll rename these and the key in the above suggestion above to something clearer.